plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -279,8 +279,8 @@ public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
     private long getModificationTime(Path path)
     {
         try {
-            FileStatus fileStatus = hdfsEnvironment.doAs(hdfsContext.getIdentity(), () -> hdfsEnvironment.getFileSystem(hdfsContext, path).getFileStatus(path));
-            return fileStatus.getModificationTime();
+            TrinoInputFile inputFile = fileSystemFactory.create(session).newInputFile(path.toString());
```

Comment:
The caller doesn't need to construct Path, make getModificationTime take a String